crates/bin/pcli/src/command/tx/auction/dutch/mod.rs

```diff
@@ -0,0 +1,448 @@
+                        AddressIndex::new(*source),
+                    )
+                    .await
+                    .context("can't build send transaction")?;
+                app.build_and_submit_transaction(plan).await?;
+                Ok(())
+            }
+            DutchCmd::DutchAuctionWithdraw {
+                source,
+                auction_id,
+                fee_tier,
+            } => {
+                let auction_id = auction_id.parse::<AuctionId>()?;
+
+                use pbjson_types::Any;
+                let view_client = app.view();
+                let (auction_id, _, auction_raw, _): (
+                    AuctionId,
+                    SpendableNoteRecord,
+                    Option<Any>,
+                    Vec<Position>,
+                ) = view_client
+                    .auctions(None, true, true)
+                    .await?
+                    .into_iter()
+                    .find(|(id, _, _, _)| &auction_id == id)
+                    .ok_or_else(|| anyhow!("the auction id is unknown from the view service!"))?;
+
+                let Some(raw_da_state) = auction_raw else {
+                    bail!("auction state is missing from view server response")
+                };
+
+                use penumbra_proto::core::component::auction::v1alpha1 as pb_auction;
+                // We're processing a Dutch auction:
+                assert_eq!(raw_da_state.type_url, pb_auction::DutchAuction::type_url());
+
+                let dutch_auction = DutchAuction::decode(raw_da_state.value)?;
+
+                let reserves_input = Value {
+                    amount: dutch_auction.state.input_reserves,
+                    asset_id: dutch_auction.description.input.asset_id,
+                };
+                let reserves_output = Value {
+                    amount: dutch_auction.state.output_reserves,
+                    asset_id: dutch_auction.description.output_id,
+                };
+                let seq = dutch_auction.state.sequence + 1;
+
+                let mut planner = Planner::new(OsRng);
+
+                let plan = planner
+                    .set_gas_prices(gas_prices)
+                    .set_fee_tier((*fee_tier).into())
+                    .dutch_auction_withdraw(ActionDutchAuctionWithdrawPlan {
+                        auction_id,
+                        seq,
+                        reserves_input,
+                        reserves_output,
+                    })
+                    .plan(
+                        app.view
+                            .as_mut()
+                            .context("view service must be initialized")?,
+                        AddressIndex::new(*source),
+                    )
+                    .await
+                    .context("can't build send transaction")?;
+                app.build_and_submit_transaction(plan).await?;
+                Ok(())
+            }
+            DutchCmd::DutchAuctionGradualSchedule {
```

Comment:
:speech_balloon: **nit:** it'd be nice if this arm was next to the other `DutchCmd::DutchAuctionSchedule` arm, without the withdraw/end arms between, since changes to one auction type will likely tie into changes in the other auction type.

Reply:
punting, "gradual" auction support is an r&d tool.